config/routes.rb

```diff
@@ -1,5 +1,14 @@
 Rails.application.routes.draw do
+  
+  root 'restaurants#index'
+
   resources :restaurants
+
+  get 'restaurants/:id/destroy' => 'posts#destroy',  as: :delete_restaurant
```

Comment:
a destroy route occurs as part of resources. You just need to change the link that's supposed to destroy a restaurant to use the proper method.